Makefile

```diff
@@ -78,5 +78,6 @@ deploy: ## Deploy cloud-api-adaptor using the operator, according to install/ove
 .PHONY: delete
 delete: ## Delete cloud-api-adaptor using the operator, according to install/overlays/$(CLOUD_PROVIDER)/kustomization.yaml file.
 	kubectl delete -k install/overlays/$(CLOUD_PROVIDER)
+	kubectl delete -f install/yamls/deploy.yaml
```

Comment:
Is this required? AFAIR we deliberately didn't remove those, cc @bpradipt

Reply:
hmmm... if it was deliberated then mind to explain the rationale? :)

Reply:
I don't recall, probably no that important, maybe @jensfr could say if it make sense

Reply:
I think the Makefile changes should be moved to a separate PR. Reason being to discuss whether we really need to install the operator in `deploy` target ? or just refer to operator instructions ?
IOW, should the `deploy` and `delete` (is `undeploy` the right target?) only cater to caa ?

Reply:
Thanks for the input @bpradipt ! I will move those makefile changes to a separate PR.